src/plugins/analysis/crypto_material/test/test_plugin_crypto_material.py

```diff
@@ -1,52 +1,63 @@
 import os
 
+import pytest
 from common_helper_files import get_dir_of_file
 
-from test.unit.analysis.AbstractSignatureTest import AbstractSignatureTest  # pylint: disable=wrong-import-order
+from objects.file import FileObject
 
 from ..code.crypto_material import AnalysisPlugin
 
+TEST_DATA_DIR = os.path.join(get_dir_of_file(__file__), 'data')
 
-class CryptoCodeMaterialTest(AbstractSignatureTest):
-    PLUGIN_NAME = 'crypto_material'
-    PLUGIN_CLASS = AnalysisPlugin
-    TEST_DATA_DIR = os.path.join(get_dir_of_file(__file__), 'data')
 
-    def test_gnupg(self):
-        self._rule_match('0x6C2DF2C5-pub.asc', 'PgpPublicKeyBlock', len(['PgpPublicKeyBlock', 'PgpPublicKeyBlock_GnuPG']))
+def _rule_match(analysis_plugin, filename, expected_rule_name, expected_number_of_rules=1):
+    path = os.path.join(TEST_DATA_DIR, filename)
+    test_file = FileObject(file_path=path)
+    analysis_plugin.process_object(test_file)
+    number_of_rules = len(test_file.processed_analysis[analysis_plugin.NAME]) - 1
+    assert number_of_rules == expected_number_of_rules, f'Number of results is {number_of_rules} but should be {expected_number_of_rules}'
+    if expected_rule_name is not None:
+        assert expected_rule_name in test_file.processed_analysis[analysis_plugin.NAME], f'Expected rule {expected_rule_name} missing'
 
-    def test_ssh_public(self):
-        self._rule_match('id_rsa.pub', 'SshRsaPublicKeyBlock')
 
-    def test_ssh_private(self):
-        self._rule_match('id_rsa', 'SshRsaPrivateKeyBlock', expected_number_of_rules=2)
+@pytest.mark.AnalysisPluginClass.with_args(AnalysisPlugin)
+class TestCryptoMaterial:
+    # pylint:disable=no-self-use
```

Comment:
Ideally I'd like to have all tests outside of a class.
But to my knowledge there is know other way to set a pytest marker for many tests at once other than putting the tests in a class.
The marker in question is `@pytest.mark.AnalysisPluginClass.with_args(AnalysisPlugin)`.

And rather than adding many lines of `@staticmethod` I think just one line of disabeling pytest lints is better.

Reply:
It seems to be possible for whole files like [this](https://developers.perfectomobile.com/display/TT/PyTest+Marks)
I wouldn't mind much if it's just in tests and IMHO we can keep it this way.

Reply:
As discussed I'll leave it like it is currently.